src/explore-education-statistics-admin/src/components/editable/__tests__/EditableContentForm.test.tsx

```diff
@@ -0,0 +1,339 @@
+              onDeletePendingComment: jest.fn(),
+              onSaveComment: jest.fn(),
+              onSaveUpdatedComment: handleUpdateComment,
+            }}
+          >
+            <EditableContentForm
+              allowComments
+              content="Test content"
+              id="block-id"
+              label="Form label"
+              onCancel={noop}
+              onSubmit={noop}
+            />
+          </CommentsProvider>
+        </AuthContext.Provider>,
+      );
+
+      const unresolvedComments = within(
+        screen.getByTestId('unresolvedComments'),
+      ).getAllByRole('listitem');
+
+      expect(unresolvedComments).toHaveLength(3);
+
+      userEvent.click(
+        within(unresolvedComments[0]).getByRole('button', {
+          name: 'Resolve',
+        }),
+      );
+
+      await waitFor(() => {
+        expect(
+          within(screen.getByTestId('unresolvedComments')).getAllByRole(
+            'listitem',
+          ),
+        ).toHaveLength(2);
+      });
+
+      expect(
+        screen.getByRole('button', {
+          name: 'Resolved comments (3)',
+        }),
+      ).toBeInTheDocument();
+
+      userEvent.click(
+        screen.getByRole('button', {
+          name: 'Resolved comments (3)',
+        }),
+      );
+
+      expect(
+        within(screen.getByTestId('resolvedComments')).getAllByRole('listitem'),
+      ).toHaveLength(3);
+    });
+
+    test('moves the comment to the unresolved list when Unresolve is clicked', async () => {
+      const handleUpdateComment = jest.fn();
+      const unresolvedComment = {
+        ...testComments[0],
+      };
+      delete unresolvedComment.resolved;
+      delete unresolvedComment.resolvedBy;
+
+      handleUpdateComment.mockResolvedValue(unresolvedComment);
+
+      render(
+        <AuthContext.Provider
+          value={{
+            user: testUser1,
+          }}
+        >
+          <CommentsProvider
+            value={{
+              comments: testComments,
+              onDeletePendingComment: jest.fn(),
+              onSaveComment: jest.fn(),
+              onSaveUpdatedComment: handleUpdateComment,
+            }}
+          >
+            <EditableContentForm
+              allowComments
+              content="Test content"
+              id="block-id"
+              label="Form label"
+              onCancel={noop}
+              onSubmit={noop}
+            />
+          </CommentsProvider>
+        </AuthContext.Provider>,
+      );
+
+      userEvent.click(
+        screen.getByRole('button', {
+          name: 'Resolved comments (2)',
+        }),
+      );
+
+      const resolvedComments = within(
+        screen.getByTestId('resolvedComments'),
+      ).getAllByRole('listitem');
+
+      expect(resolvedComments).toHaveLength(2);
+
+      userEvent.click(
+        within(resolvedComments[0]).getByRole('button', {
+          name: 'Unresolve',
+        }),
+      );
+
+      await waitFor(() => {
```

Comment:
Should we be using `screen.getByText` here in order to be more performant? (see https://github.com/dfe-analytical-services/explore-education-statistics/pull/3053)

Reply:
Yep good catch, sorted now.